frontend/packages/operator-lifecycle-manager/src/components/operand/index.tsx

```diff
@@ -330,6 +334,8 @@ export const ProvidedAPIsPage = (props: ProvidedAPIsPageProps) => {
   const { t } = useTranslation();
   const { obj } = props;
   const [models, inFlight] = useK8sModels();
+  const dispatch = useDispatch();
+  React.useEffect(() => dispatch(getResources()), [dispatch]);
```

Comment:
Can we skip discovery if all the models we need are there?

Reply:
The only problem is that running API discovery doesn't guarantee that the models will be found, so we can end up in a situation where API discovery just keeps getting run over and over. Unless we make an exception to the exhaustive dependency rule and only run this effect once?

Reply:
We could wait until `inFlight` is false, then check if all models are present If not, run discovery once. So we wouldn't need exhaustive deps, just `[inFlight, dispatch]`.